src/interiorpointsolver.jl

```diff
@@ -265,9 +265,8 @@ function eval_grad_f_wrapper!(ipp::Solver, f::Vector{Float64},x::Vector{Float64}
     nlp = ipp.nlp
     cnt = ipp.cnt
     @trace(ipp.logger,"Evaluating objective gradient.")
-    ∇f = view(f, 1:nlp.n)
-    cnt.eval_function_time += @elapsed nlp.obj_grad!(∇f,view(x,1:nlp.n))
-    ∇f.*=ipp.obj_scale[]
+    cnt.eval_function_time += @elapsed nlp.obj_grad!(f,view(x,1:nlp.n))
+    f.*=ipp.obj_scale[]
     cnt.obj_grad_cnt+=1
```

Comment:
The issue was here; for some reason, sending `∇f` caused some extra allocations. Maybe the reason comes from CUTEst.jl. If the gradient callback checks the dimension of `f`, the current change can be problematic

Reply:
For a similar reason, if we replace `view(x,1:nlp.n)` with `x`, allocations are significantly reduced. But if the gradient callback checks the input size, it can cause issue.

Reply:
Indeed, the issue was here:
https://github.com/JuliaSmoothOptimizers/CUTEst.jl/blob/main/src/julia_interface.jl#L105-L110

In order to make it C function compatible, CUTEst.jl internally converts the inputs to `Vector{Float64}`. For "EIGMINA" in CUTEst, after allowing StrideOneVector{Float64} in the inputs, the allocation dramatically reduces:
from 1.213 KiB to 759.922 KiB. 

I'll create a PR for CUTEst.jl. I think the benchmark result will look a lot better after this...

Reply:
That's very interesting. Good catch for the issue with CUTest.jl. I think indeed there is no need to create a new array if the input `x` is one-strided ... I did this change because of the custom wrapper we developed for ExaPF: here, we need a gradient with the correct dimension (as we are using `mul!` operation afterwards to compute the adjoint). I don't know what's the default: should we pass to the callbacks all arrays with the correct dimension, or should the callbacks do all the checks internally?

Reply:
I think the latter would be ideal, but the performance of MadNLP shouldn't degrade too much even if the callback requires the correct dimension. I added a "buffered" option for NLPModels.jl interface so that there is a buffer vector that exactly matches the input dimensions for the callbacks. NLPModels.jl interface handles copying to/from the buffer vector. But if the user knows that there won'te be a problem, the user can turn this off.

I made a PR for CUTEst.jl (https://github.com/JuliaSmoothOptimizers/CUTEst.jl/pull/269) to allow view inputs. If this gets merged, we can turn off buffered option for the benchmark.